components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx

```diff
@@ -20,45 +30,124 @@ const withLiveBlogWrapperActions = withActions({
 	}
 })
 
-const BaseLiveBlogWrapper = ({
-	posts = [],
-	ads = {},
-	articleUrl,
-	showShareButtons,
-	id,
-	liveBlogWrapperElementRef
-}) => {
-	posts.sort((a, b) => {
-		const timestampA = a.publishedDate || a.publishedTimestamp
-		const timestampB = b.publishedDate || b.publishedTimestamp
-
-		// Newer posts on top
-		if (timestampA > timestampB) {
-			return -1
+class BaseLiveBlogWrapper extends Component {
+	/**
+	 * @param {Object} props
+	 * @param {Object[]} props.post - A list of liveblog posts
+	 * @param {Object} props.ads
+	 * @param {string} props.articleUrl - The base url for the article
+	 * @param {boolean} props.showShareButtons - condition to show shareButtons
+	 * @param {string} props.id - The id of the liveblog package
+	 * @param {*} props.liveBlogWrapperElementRef
+	 * @param {PostTrackerConfig} props.postTrackerConfig - Optional config for tracking post
+	 */
+	constructor(props) {
+		super(props)
+	}
+
+	componentDidMount() {
+		this.setUp()
+	}
+
+	componentWillUnmount() {
+		if (this.state.tracker) {
+			this.state.tracker.destroy()
+		}
+	}
+
+	setUp() {
+		const { onEntersViewport, onRead, onError, usePostTracker } = this.props.postTrackerConfig || {}
+
+		if (!usePostTracker) {
+			return
 		}
 
-		if (timestampB > timestampA) {
-			return 1
+		if (
+			typeof onEntersViewport !== 'function' ||
+			typeof onRead !== 'function' ||
+			typeof onError !== 'function'
+		) {
+			// eslint-disable-next-line no-console
+			console.error(
+				'onEntersViewport, onRead and onError callback functions are required to use Post tracker'
+			)
+			return
+		}
+
+		const tracker = this.setUpPostTracking({ onEntersViewport, onRead, onError }, this.props.id)
+		this.setState({
+			tracker
+		})
+	}
+
+	/**
+	 *
+	 * @param {Object} callbacks - Callbacks to be called by PostTracker class
+	 * @param {Function} callbacks.onEntersViewport - function to be called when the criteria for `view` in Post tracker has been fulfiled
+	 * @param {Function} callbacks.onRead - function to be called when the criteria for `read` in Post tracker has been fulfiled
+	 * @param {Function} callbacks.onError - function to be called when PostTracker encounters an error
+	 * @param {string} id
+	 * @returns {PostTracker}
+	 */
+	setUpPostTracking(callbacks, id) {
+		/**
+		 * @type {import('./lib/post-tracker').PostTrackerConfig}
+		 */
+		let config = {
+			query: 'article[data-trackable="live-post"]',
+			minMillisecondsToReport: 5000,
+			returnVisibleElement: true,
+			observerUpdateEventString: 'LiveBlogWrapper.INSERT_POST',
+			liveBlogWrapperQuery: `div[data-live-blog-wrapper-id="${id}"]`,
+			liveBlogWrapper: this.props.liveBlogWrapperElementRef
+				? this.props.liveBlogWrapperElementRef.current
+				: undefined,
+			onEntersViewport: (event) => callbacks.onEntersViewport(event),
+			onRead: (event) => callbacks.onRead(event),
+			onError: (event) => callbacks.onError(event)
```

Comment:
**Suggestion/Feedback:** This could error if a consuming app doesn't pass these properties in as they won't be functions. I think we need some sanitising of whats passed in to make sure we can set up a post tracker

Reply:
So I wrote that as a `sub function` to the `setup function`. The setup function does the sanitisation here https://github.com/Financial-Times/x-dash/blob/bbe48a46699f09ca60c92a3a56e89273c83f0ded/components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx#L65 and errors out here https://github.com/Financial-Times/x-dash/blob/bbe48a46699f09ca60c92a3a56e89273c83f0ded/components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx#L71.

Would you prefer the sanitisation inside the sub function? I left all sanitisation inside the setup function.

Reply:
I knew I had probably missed something 😄 Sorry that looks fine - just hard to follow it in the PR